python/datafusion/context.py

```diff
@@ -0,0 +1,1167 @@
+
+        Returns
+        -------
+        DataFrame
+            DataFrame representation of the Pandas DataFrame.
+        """
+        return DataFrame(self.ctx.from_pandas(data, name))
+
+    def from_polars(self, data: polars.DataFrame, name: str | None = None) -> DataFrame:
+        """Create a `DataFrame` from a Polars DataFrame.
+
+        Parameters
+        ----------
+        data : polars.DataFrame
+            Polars DataFrame.
+        name : str | None
+            Name of the DataFrame.
+
+        Returns
+        -------
+        DataFrame
+            DataFrame representation of the Polars DataFrame.
+        """
+        return DataFrame(self.ctx.from_polars(data, name))
+
+    def register_table(self, name: str, table: pyarrow.Table) -> None:
+        self.ctx.register_table(name, table)
+
+    def deregister_table(self, name: str) -> None:
+        self.ctx.deregister_table(name)
+
+    def register_record_batches(
+        self, name: str, partitions: list[list[pyarrow.RecordBatch]]
+    ) -> None:
+        self.ctx.register_record_batches(name, partitions)
+
+    def register_parquet(
+        self,
+        name: str,
+        path: str,
+        table_partition_cols: list[tuple[str, str]] = [],
+        parquet_pruning: bool = True,
+        file_extension: str = ".parquet",
+        skip_metadata: bool = True,
+        schema: pyarrow.Schema | None = None,
+        file_sort_order: list[list[Expr]] | None = None,
+    ) -> None:
+        """Register a Parquet file as a table.
+
+        The registered table can be referenced from SQL statement executed against
+        this context.
+
+        Parameters
+        ----------
+        name : str
+            Name of the table to register.
+        path : str
+            Path to the Parquet file.
+        table_partition_cols : list[tuple[str, str]], optional
+            Partition columns, by default []
+        parquet_pruning : bool, optional
+            Whether the parquet reader should use the predicate to prune row groups, by default True
+        file_extension : str, optional
+            File extension; only files with this extension are selected for data input, by default ".parquet"
+        skip_metadata : bool, optional
+            Whether the parquet reader should skip any metadata that may be in the file
+            schema. This can help avoid schema conflicts due to metadata. by default True
+        schema : pyarrow.Schema | None, optional
+            The data source schema, by default None
+        file_sort_order : list[list[Expr]] | None, optional
+            Sort order for the file, by default None
+        """
+        self.ctx.register_parquet(
+            name,
+            path,
+            table_partition_cols,
+            parquet_pruning,
+            file_extension,
+            skip_metadata,
+            schema,
+            file_sort_order,
+        )
+
+    def register_csv(
+        self,
+        name: str,
+        path: str,
```

Comment:
Same comment as above about `pathlib.Path`

Reply:
Updated and also unit test coverage.